### ISSUE TITLE
Add a module to handle possible failure from the Reader.

### ISSUE THREAD

**Issue by Punie** [Punie/elm-reader#1]
I would like to add a module that would provide some helpful functions to handle possible failure (in the form of a `Result e a` from the Reader).

Working with nested ADT can be cumbersome and lead to heavily nested case statement and use of various specialised `|> andThen` operators so providing some helper functions to ease the manipulation of the expected value through the stack can be useful.

Basically a type alias for `Reader env (Result err value)` should do the trick. It would provide somewhat equivalent functionality to the `ExceptT e m a` monad transformer from Haskell specialised for the Reader type from this package.

**Comment by Punie** [comment]
Closed by PR #2
